File: src/main/java/com/example/fit_tracker/service/ActivityService.java
package com.example.fit_tracker.service;



import com.example.fit_tracker.dto.ActivityDTO;

import java.util.List;

public interface ActivityService {
    ActivityDTO postActivity(ActivityDTO dto);

    List<ActivityDTO> getActivities();

    ActivityDTO updateActivity(Long id, ActivityDTO dto); // Изменен Object на ActivityDTO
    String deleteActivity(Long id);
}

File: fit-tracker-frontend/src/api/ActivityService.js

import api from './axiosConfig';

export const ActivityService = {

    getActivities: async () => {
        const response = await api.get('/activities');
        return response.data;
    },

    postActivity: async (activityData) => {
        const response = await api.post('/activity', activityData);
        return response.data;
    },

    // Обновить существующую запись
    updateActivity: async (id, activityData) => {
        const response = await api.put(`/activity/${id}`, activityData);
        return response.data;
    },

    deleteActivity: async (id) => {
        const response = await api.delete(`/activity/${id}`);
        return response.data; // Ожидаем сообщение об успехе (String)
    },
};